lib/src/connector/connector.cc

```diff
@@ -317,7 +326,13 @@ void Connector::stopMonitoring()
     checkConnectionInitialization();
 
     if (!is_monitoring_) {
-        LOG_WARNING("The Monitoring Thread is not running");
+        if (!monitor_exception_) {
```

Comment:
This set of if-statements might be cleaner if it mirrored the logic in the destructor.